### PR TITLE
feat: Expand Test Coverage for execute_token_transaction

### LINKED ISSUE #142: [TEST] Expand Test Coverage for `execute_token_transaction`
### **Description**

The current test suite for the `execute_token_transaction` functionality only covers the successful execution path. To ensure robustness and prevent regressions, we need to implement test cases for relevant **edge cases** and verify **event emissions** throughout the transaction lifecycle — from proposal to execution.

---

### **Edge Cases to Cover**

1. **Non-existent or invalid transaction**
   Revert if the transaction:

   * Does not exist
   * Exists but is not a token transaction

2. **Insufficient approvals**
   Revert if the number of approvals has not met the required threshold.

3. **Unauthorized caller**
   Revert if the caller is not an authorized executor.

4. **Insufficient token balance**
   Revert if the account does not have enough tokens to complete the transaction.

5. **Zero address recipient**
   Revert if the recipient address is the zero address 

---

### **Event Emissions to Verify**

Ensure the following events are emitted correctly:

* `TransactionApproved`: When an approval is added
* `TransactionExecuted`: When the transaction is successfully executed
* `TokenTransactionExecuted`: When the token transaction is executed

---

### **Proposed Steps**

1. Update the test file (e.g., `test_token_transaction`) to include new tests.

2. Add separate tests for each of the edge cases listed above.

3. Add tests that simulate the full lifecycle:

   * Propose → Approve → Execute
   * At each stage, validate that correct events are emitted with expected parameters

4. Use the testing framework's event assertion methods (`spyEvent`) to validate emitted events.

---

### **Acceptance Criteria**

* [ ] Tests exist for all specified edge cases
* [ ] Each test raises the proper errors
* [ ] All relevant events are asserted at the proposal, approval, and execution stages
* [ ] The test suite passes successfully and maintains/increases code coverage
* [ ] No existing functionality is broken by the new tests

### PR DESCRIPTION
**Pull request by anihdev**
## Description 📝
This PR expands test coverage for `execute_token_transaction` functionality, adding edge cases and event validation as specified in #142. 

## Key improvements include:

- Updated error message for clarity: `ERR_INVALID_TOKEN_TRANSACTION` to `ERR_INVALID_TOKEN_TRANSACTION_TYPE`, adjusted all contract references to use new error name.  
- Ensured consistent error handling across components
- Added comprehensive event validation using `assert_emitted`
- Covered all specified edge cases with focused tests

## Tests Added 

- `test_propose_token_transaction_fail_zero_recipient` - Verifies zero recipient address rejection  
- `test_execute_token_transaction_fail_non_existent` - Tests invalid transaction ID handling  
- `test_execute_token_transaction_fail_invalid_type` - Ensures only token transactions execute  
- `test_execute_token_transaction_fail_insufficient_approvals` - Validates threshold enforcement  
- `test_execute_token_transaction_fail_unauthorized_executor` - Checks executor permissions  
- `test_execute_token_transaction_fail_insufficient_balance_at_execution` - Tests balance checks  
- `test_execute_token_transaction_fail_already_executed` - Prevents double execution  
- `test_propose_token_transaction_fail_zero_amount` - Rejects zero amount transfers  
- `test_pause_contract_prevents_execution` - Validates pause functionality  
- `test_full_token_transaction_lifecycle_with_events` - overall flow with event validation  
- `test_approve_transaction_event_emission` - Verifies approval events  
- `test_multiple_approvals_before_execution` - Tests multi-sign scenarios  

## Related Issues 🔗
 Fixes #142.

## Changes Made 🚀
- [ ✅] ✨ Feature Implementation 
- [ ✅] 🐛 Bug Fix 
- [ ✅] 📚 Documentation Update 
- [✅ ] 🔨 Refactoring 

## Checklist ✅
- [ ✅] 🛠 I have tested these changes. 
- [ ✅] 📖 I have updated the documentation.
- [ ✅] 🎨 This PR follows the project's coding style. 
- [✅ ] 🧪 I have added necessary tests. 

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated error message for invalid token transactions to provide clearer feedback.

* **Tests**
  * Significantly expanded token transaction test coverage, including new scenarios for failure cases, event emissions, permission checks, and contract pause behavior.
  * Added helper functions to streamline test setup and simulate different user roles.

* **Style**
  * Added a comment for clarification in an existing test case.

* **Documentation**
  * Improved visibility of event fields in transaction-related events for better transparency.

### CODE REVIEW

File: src/tests/test_spherre.cairo [hunk 1]
```diff
@@ -739,7 +739,7 @@ fn test_update_account_class_hash_emits_event() {
     );
 
     spy.assert_emitted(@array![(spherre_contract, expected_event)]);
-}
+} //@note
```
Comment (codebestia) [inline src/tests/test_spherre.cairo:742]:
```suggestion
}
```

**Review by codebestia — APPROVED** [review]
LGTM!
Thank you for your contribution.
